pkg/controller/acmechallenges/sync.go

```diff
@@ -260,19 +253,8 @@ func (c *controller) handleFinalizer(ctx context.Context, ch *cmacme.Challenge)
 	}
 
 	defer func() {
-		// call UpdateStatus first as we may have updated the challenge.status.reason field
-		ch, updateErr := c.updateStatusOrApply(ctx, ch)
```

Comment:
The original code unconditionally updated the status....the comment "as we may have updated the challenge.status.reason" seems dubious as there's nothing executed before this that might update the reason field AFAICS.

Now the updateObject function will invoke UpdateStatus if any of the Status fields have changed.

Reply:
Also this line is interesting- I did not realize that we _always_ remove the finalizer even if the clean up failed. I seem to remember seeing some issues where people compain that the resources are not cleaned up - perhaps this is why. I wonder if there is some reason why we do this.

Reply:
That's why I linked this PR to https://github.com/cert-manager/cert-manager/issues/3640